pydantic/main.py

```diff
@@ -282,6 +282,12 @@ def is_untouched(v: Any) -> bool:
         cls = super().__new__(mcs, name, bases, new_namespace, **kwargs)
         # set __signature__ attr only for model class, but not for its instances
         cls.__signature__ = ClassAttribute('__signature__', generate_model_signature(cls.__init__, fields, config))
+
+        if not _is_base_model_class_defined:
+            # Cython does not understand the `if TYPE_CHECKING:` condition in the
+            # BaseModel's body (where annotations are set), so clear them manually:
+            getattr(cls, '__annotations__', {}).clear()
+
```

Comment:
Why is this necessary? Maybe I'm missing context above...

Reply:
Not sure how Cython is handling this, but turns out it is not related to `typing.TYPE_CHECKING`. Any condition in the class body is discarded, as long as it is not directly using the `False` literal, e.g.:

```python
var = False

class A:
    if var:
        ann: int

A.__annotations__
#> {'ann': int}

Reply:
Alas, I guess we just go with what you have then.